source/main/resources/CacheSystem.cpp

```diff
@@ -934,11 +933,38 @@ void CacheSystem::fillTruckDetailInfo(CacheEntry& entry, Ogre::DataStreamPtr str
     entry.turbojetcount = static_cast<int>(def->root_module->turbojets.size());
     entry.flexbodiescount = static_cast<int>(def->root_module->flexbodies.size());
     entry.soundsourcescount = static_cast<int>(def->root_module->soundsources.size() + def->root_module->soundsources.size());
-    entry.wheelcount = static_cast<int>(
-        def->root_module->wheels.size()
-        + def->root_module->wheels_2.size()
-        + def->root_module->mesh_wheels.size() // Also meshwheels2
-    );
+
+    entry.wheelcount = 0;
+    entry.propwheelcount = 0;
+    for (const auto& w : def->root_module->wheels)
+    {
+        entry.wheelcount++;
+        if (w.propulsion != RigDef::Wheels::PROPULSION_NONE)
+            entry.propwheelcount++;
+    }
+    for (const auto& w : def->root_module->wheels_2)
+    {
+        entry.wheelcount++;
+        if (w.propulsion != RigDef::Wheels::PROPULSION_NONE)
+            entry.propwheelcount++;
+    }
+    for (const auto& w : def->root_module->mesh_wheels)
+    {
+        entry.wheelcount++;
+        if (w.propulsion != RigDef::Wheels::PROPULSION_NONE)
+            entry.propwheelcount++;
+    }
+    for (const auto& w : def->root_module->flex_body_wheels)
+    {
+        entry.wheelcount++;
+        if (w.propulsion != RigDef::Wheels::PROPULSION_NONE)
+            entry.propwheelcount++;
+    }
+
+    if (!def->root_module->axles.empty())
+    {
+        entry.propwheelcount = static_cast<int>(def->root_module->axles.size() * 2);

```

Comment:
Wheels can have both 'propulsion' attr set and be part of an axle, axle then overrides the attr, so this needs an extra check.

Reply:
`=`, not `+=` though :thinking:

Reply:
> Does anybody even read that value in the selector?

It was requested by the community.